sky/serve/load_balancer.py

```diff
@@ -240,11 +285,286 @@ def run_load_balancer(controller_addr: str,
+            }]
+        }
+
+        # A cluster is usually a group of endpoints that can be load balanced
+        # over.  This one says to find the list of endpoints in eds.yaml.
+        cluster = {
+            'name': 'cluster',
+            'connect_timeout': '0.25s',
+            'type': 'EDS',
+            'lb_policy': 'ROUND_ROBIN',
+            'eds_cluster_config': {
+                'eds_config': {
+                    'path_config_source': {
+                        'path': '/etc/envoy/eds.yaml'
+                    }
+                }
+            }
+        }
+
+        config = {
+            'node': {
+                'id': 'controller',
+                'cluster': self._service_name,
+            },
+            'static_resources': {
+                'listeners': [listener],
+                'clusters': [cluster],
+            }
+        }
+
+        os.makedirs(self.envoy_config_folder, exist_ok=True)
+        self.write_yaml(config, 'envoy.yaml')
+
+    def write_eds(self, replicas: List[str]):
+        """ Writes eds.yaml
+
+        Endpoint Discovery Service (EDS) is a subset of Envoy xds that's used
+        for discovering backend load balancing endpoints for a particular
+        envoy cluster (load balancer).  In our case, this contains the list of
+        replicas that our service can route to.
+
+        This method writes eds.yaml to the correct location.  It's intended to
+        be called every time the set of replicas changes.
+        """
+
+        lb_endpoints = []
+        for url in replicas:
+            # TODO(ejj) it would be cleaner if the controller just sent us
+            # tuple containing IP and port rather than a url which we have to
+            # parse here.
+            parsed_url = urlparse(url)
+            lb_endpoints.append({
+                'endpoint': {
+                    'address': {
+                        'socket_address': {
+                            'address': parsed_url.hostname,
+                            'port_value': parsed_url.port,
+                        }
+                    }
+                }
+            })
+
+        config = {
+            'resources': {
+                '@type': 'type.googleapis.com/envoy.config'
+                         '.endpoint.v3.ClusterLoadAssignment',
+                'cluster_name': 'cluster',
+                'endpoints': {
+                    'lb_endpoints': lb_endpoints
+                }
+            }
+        }
+        self.write_yaml(config, 'eds.yaml')
+
+    async def run(self):
+        self.write_bootstrap_xds()
+
+        # Because docker can take some time to come up, we make multiple
+        # attempts to start before giving up.
+        envoy_started = False
+        logger.info('Starting envoy %s', self.envoy_name)
+        for _ in range(30):
+            await asyncio.sleep(5)
+            envoy_started = await self._start_envoy()
+            if envoy_started:
+                break
```

Comment:
```suggestion
        for _ in range(50):
            await asyncio.sleep(3)
            envoy_started = await self._start_envoy()
            if envoy_started:
                break
```
use finer granularity? 5 seconds looks a little bit long for me.